src/receive.rs

```diff
@@ -1364,8 +1364,13 @@ impl DiscoveredSacnSource {
     pub fn has_all_pages(&mut self) -> bool {
         // https://rust-lang-nursery.github.io/rust-cookbook/algorithms/sorting.html (31/12/2019)
         self.pages.sort_by(|a, b| a.page.cmp(&b.page));
-        for i in 0..(self.last_page + 1) {
-            if self.pages[i as usize].page != i {
+        for i in 0..=self.last_page {
```

Comment:
```suggestion
        for i in 0..=(self.last_page+1) {
```

Is there a fencepost error introduced by this change? If lastpage is also an index, a..b iterates a<b not <=b.

Reply:
I changed the non-inclusive `last_page+1` to the inclusive `=last_page`, since it should follow the same logic, and I think it shows the iteration range intention more clearly. Happy to change it back to `0..(self.last_page+1)` if that reads better.